test/test_torch.py

```diff
@@ -3094,59 +3094,54 @@ def _test_broadcast(self, cast):
                 # map and map2 are not implementd on CUDA tensors
                 continue
 
-            # TODO: fix masked_scatter and masked_fill broadcasting
-            if hasattr(large_expanded, fn) and fn not in ['masked_scatter', 'masked_fill']:
+            def specialized(myfn, t1, t2, t3):
+                if fn == "lerp":
+                    return myfn(t1, t2, 0.5)
+                elif fn == "masked_select":
+                    return myfn(t1, t2 < 0)
+                elif fn == "masked_scatter":
+                    expanded = torch.broadcast_tensors(t1, t2)[0]
+                    return myfn(t1, t2 < 0.5, torch.ones_like(expanded))
+                elif fn == "masked_fill":
+                    return myfn(t1, t2 < 0.5, 1.0)
+                elif fn in fns_3_args:
+                    return myfn(t1, 1.0, t2, t3)
+                else:
+                    return myfn(t1, t2)
+
+            if hasattr(large_expanded, fn):
                 # run through tensor versions of functions
                 # and verify fully expanded inputs give same results
                 expanded = {large: large_expanded, small: small_expanded, small2: small2_expanded}
 
-                def tensorfn(myfn, t1, t2):
-                    if fn == "lerp":
-                        return myfn(t1, 0.5)
-                    elif fn == "masked_select":
-                        return myfn(t1 < 0)
```

Comment:
`masked_fill` and `masked_scatter` is not specialized here, because these two operators are explicitly excluded by the `if hasattr(large_expanded, fn) and fn not in ['masked_scatter', 'masked_fill']:` above, but now since it is no longer excluded, we need to specialize these two operators here as in the `def torchfn` below. Then this function and `def torchfn` become duplicate, so I merge them and name it `specialized`